### PR TITLE
Add support for SSL connections

### PR DESCRIPTION
**Pull request by mlockett42**
This PR adds support for connecting to an EOS node over HTTPS. I does this by adding the protocol as one of the members for the Network model for EOS networks. This was judged to be the only effective way of implementing SSL support because of security reasons.

Networks created with previous version of this software should upgrade automatically to use the http protocol.

Ethereum networks should continue to work in the old way.

This resolves issue #72

### CODE REVIEW

File: src/models/Network.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import {Blockchains} from './Blockchains';
 
 export default class Network {
-    constructor(_host = '', _port = 0, blockchain = Blockchains.EOS, chainId = ''){
+    constructor(_host = '', _port = 0, blockchain = Blockchains.EOS, chainId = '', _protocol='http'){
```
Comment (kesar) [inline src/models/Network.js]: if this change doesnt affect to many places, I would place the _protocol as first item (and change that everywhere).

File: src/views/NetworkView.vue [hunk 2]
```diff
@@ -9,6 +9,7 @@
             <figure class="header">{{locale(langKeys.NETWORK_Header)}}</figure>
             <figure class="sub-header">{{locale(langKeys.NETWORK_Description)}} </figure>
             <sel :selected="blockchains[0]" :options="blockchains" :parser="blockchain => blockchain.key" v-on:changed="changed => bind(changed.value, 'blockchain')"></sel>
+            <sel :selected="protocols[0]" :options="protocols" :parser="protocol => protocol.key" v-on:changed="changed => bind(changed.value, 'protocol')" v-if="network.blockchain===eos"></sel>
```
Comment (kesar) [inline src/views/NetworkView.vue]: do we need **network.blockchain===eos** ?

**Comment by nsjames** [comment]
I just landed. Let me get situated and I'll check the implications of this.

Technically scatter should run protocol-less internally for importation and ABI parsing however eosjs doesn't seem to support that. 

A better fix might be removing the port dependency for account finding when requesting identities since for the actual `scatter.eos` call it shouldn't matter the network provided. I'll deep dive this tonight and hopefully push a fix.


On another note, if this is done this way it'll need a migration to update all networks to support it which isn't a big deal either.

File: src/models/Network.js (hunk 1 above)
Comment (mlockett42) [reply src/models/Network.js]: At first when I was writing this that's what I did. But when I looked at the Ethereum support I got a bit paranoid at the changes I would be making. The Ethereum support doesn't use that parameter and I don't reallyuuuu understand how ethereum works at that level.

There also wasn't any code I could I test against. Ie any working Ethereum example apps.

This line is one such example.
https://github.com/EOSEssentials/Scatter/blob/6c380b2bcc2b8b0907686de3b6a7a9b5d6e550f2/src/plugins/defaults/eth.js#L119

On the other hand if you guys don't think this change will screw or the Ethereum support you are right the protocol make a lot more sense as the first parameter

**Comment by mlockett42** [comment]
@nsjames One problem with port less would be that Scatter needs to connect to the API end point to get the account name when linking the identity.

In the longer term 'normie' end users should never see those API end points. They should just connect to EOS Mainnet

**Comment by nsjames** [comment]
This is now fixed with d95c3b719ed9d7acca3c3a59fd7eeb731e9262bd which added a `protocol` to the network.

Backwards compatibility for the old `scatter.eos(..., 'https')` still exists for now.

**Pull request closed without merging**